### PR TITLE
Update Rust to 1.12.0-nightly (545a3a94f 2016-08-04)

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12712)

### CODE REVIEW

File: components/style/values/specified/mod.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use cssparser::{self, Parser, ToCss, Token};
 use euclid::size::Size2D;
 #[cfg(feature = "gecko")]
 use gecko_bindings::ptr::{GeckoArcPrincipal, GeckoArcURI};
-use parser::{ParserContext, ParserContextExtraData};
```
Comment (SimonSapin) [inline components/style/values/specified/mod.rs]: I think you’ll want `#[cfg(feature = "gecko")] use parser::ParserContextExtraData;` here.

**Comment by jdm** [comment]
```
note: C:\buildbot\slave\windows-dev\build\target\debug\deps\libsimd-823390c887f97613.rlib(simd-823390c887f97613.0.o): In function `simd::x86::sse2::{{impl}}::max':
C:/msys64/home/Administrator/.cargo\git\checkouts\simd-a428b8cc9c1d1986\master\src\x86/sse2.rs:259: undefined reference to `llvm.x86.sse2.pmaxs.w'
C:\buildbot\slave\windows-dev\build\target\debug\deps\libsimd-823390c887f97613.rlib(simd-823390c887f97613.0.o): In function `simd::x86::sse2::{{impl}}::min':
C:/msys64/home/Administrator/.cargo\git\checkouts\simd-a428b8cc9c1d1986\master\src\x86/sse2.rs:263: undefined reference to `llvm.x86.sse2.pmins.w'
C:\buildbot\slave\windows-dev\build\target\debug\deps\libsimd-823390c887f97613.rlib(simd-823390c887f97613.0.o): In function `simd::x86::sse2::{{impl}}::max':
C:/msys64/home/Administrator/.cargo\git\checkouts\simd-a428b8cc9c1d1986\master\src\x86/sse2.rs:317: undefined reference to `llvm.x86.sse2.pmaxu.b'
C:\buildbot\slave\windows-dev\build\target\debug\deps\libsimd-823390c887f97613.rlib(simd-823390c887f97613.0.o): In function `simd::x86::sse2::{{impl}}::min':
C:/msys64/home/Administrator/.cargo\git\checkouts\simd-a428b8cc9c1d1986\master\src\x86/sse2.rs:321: undefined reference to `llvm.x86.sse2.pminu.b'
collect2.exe: error: ld returned 1 exit status
```

**Comment by KiChjang** [comment]
@bors-servo retry
* #12732

**Comment by jdm** [comment]
I don't think retrying this PR will make a difference; #12732 doesn't seem to be intermittent.

**Comment by KiChjang** [comment]
Is blocked on rust-lang/rust#35300.

**Pull request merged**